backend/src/config/config.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import * as dotenv from 'dotenv';
+import * as fs from 'fs';
+import * as Joi from 'joi';
+
+export interface EnvConfig {
+  [prop: string]: string;
+}
+
+@Injectable()
+export class ConfigService {
+  private readonly envConfig: EnvConfig;
+
+  constructor() {
+    try {
+      const config = dotenv.parse(fs.readFileSync('.env'));
+      this.envConfig = this.validateInput({ ...config, ...process.env });
+    } catch (e) {
+      this.envConfig = this.validateInput(process.env);
+    }
+  }
+
+  get(key: string): string {
+    return this.envConfig[key];
+  }
+
+  get applicationPort(): string {
```

Comment:
this is the method to write getter methods so that we can easily call the config variable without invoking a method call.
now I can call `config.applicationPort` instead of `config.applicationPort()` thats the difference here.

Reply:
Okay, didn't know that syntactical structure. Thanks